core/point/point_test.go

```diff
@@ -17,121 +18,168 @@ func TestCreate(t *testing.T) {
+		},
+		{
+			desc:         "Error: Point => SGF: Y value < 0",
+			in:           New(33, -8),
+			expErrSubstr: "Point Y value < 0, out of range",
+		},
+		// Out of range X Y values
+		{
+			desc:         "Error: Point => SGF: X value > 51",
+			in:           New(53, 8),
+			expErrSubstr: "Point X value > 51, out of range",
+		},
+		{
+			desc:         "Error: Point => SGF: Y value > 51",
+			in:           New(33, 53),
+			expErrSubstr: "Point Y value > 51, out of range",
+		},
 	}
 
 	for _, tc := range testToSGFCases {
 		t.Run(tc.desc, func(t *testing.T) {
-			toSGFOut, _ := New(tc.in.x, tc.in.y).ToSGF()
-			if toSGFOut != tc.want {
+			sgfOut, err := New(tc.in.x, tc.in.y).ToSGF()
+			cerr := errcheck.CheckCases(err, tc.expErrSubstr)
+			if sgfOut != tc.want {
 				t.Errorf("%q.ToSGF() = %q, but wanted %q", tc.in,
-					toSGFOut, tc.want)
+					sgfOut, tc.want)
+				// if cerr != nil && sgfOut != tc.want {
+				if cerr != nil {
+					t.Fatal(cerr)
+				}
+				if err != nil {
+					return
+				}
+				if sgfOut == "" {
+					t.Fatal("unexpected empty SGF string value")
+				}
 			}
 		})
 	}
 }
 
 func TestSGFToPointTranslate(t *testing.T) {
 	testToPointCases := []struct {
-		desc string
-		in   string
-		want *Point
+		desc         string
+		in           string
+		want         *Point
+		expErrSubstr string
 	}{
 		{
 			desc: "SGF => Point",
 			in:   "iq",
-			want: &Point{
-				x: 8,
-				y: 16,
-			},
+			want: New(8, 16),
 		},
 		{
 			desc: "SGF => Point",
 			in:   "mf",
-			want: &Point{
-				x: 12,
-				y: 5,
-			},
+			want: New(12, 5),
 		},
 		{
 			desc: "SGF => Point",
 			in:   "Hi",
-			want: &Point{
-				x: 33,
-				y: 8,
-			},
+			want: New(33, 8),
 		},
 		{
 			desc: "SGF => Point",
 			in:   "OZ",
-			want: &Point{
-				x: 40,
-				y: 51,
-			},
+			want: New(40, 51),
+		},
+		// Error Cases:
+		// Empty string
+		{
+			desc:         "Error: SGF => Point empty string",
+			in:           "",
+			expErrSubstr: "Error: SGF X Y string values empty",
+		},
+		// Short string
+		{
+			desc:         "Error: SGF => Point short string",
+			in:           "Q",
+			expErrSubstr: "SGF X or Y string value missing",
+		},
+		// Long string
+		{
+			desc:         "Error: SGF => Point long string",
+			in:           "xyZ",
+			expErrSubstr: "SGF X Y string values empty",
 		},
 	}
 
 	for _, tc := range testToPointCases {
 		t.Run(tc.desc, func(t *testing.T) {
-			toPointOut, _ := NewFromSGF(tc.in)
-			// include the point.go *Point type X Y getters below
-			pointX := toPointOut.X()
-			pointY := toPointOut.Y()
-			if pointX != tc.want.x {
-				t.Errorf("%q.toPointOut.x = %q, but wanted %q", tc.in,
-					pointX, tc.want.x)
-			}
-			if pointY != tc.want.y {
-				t.Errorf("%q.toPointOut.y = %q, but wanted %q", tc.in,
-					pointY, tc.want.y)
+			pntOut, err := NewFromSGF(tc.in)
+			cerr := errcheck.CheckCases(err, tc.expErrSubstr)
+			// Utilizing the point.go *Point type X Y getters below
+			if pntOut != tc.want {
```

Comment:
You can't compare points this way since points are Pointer types. you have to compare the x & y values (or make an equals method). This will just compare the actual address, and will always be false.

Reply:
I actually do know that, and just missed that honestly. Also, I am having some trouble making this test cases method work, maybe I need to revisit the mechanics? And I've been very busy with school. Thanks for being flexible.